payroll/views.py

```diff
@@ -90,46 +117,84 @@ def get_context_data(self, **kwargs):
             source_link = None
 
         context.update({
-            'jobs': Job.of_employer(self.object.id, n=5),
-            'median_salary': self.median_entity_salary(),
+            'jobs': self.jobs,
+            'median_salary': self.median_entity_salary,
             'headcount': len(employee_salaries),
             'total_expenditure': sum(employee_salaries),
-            'salary_percentile': self.salary_percentile(),
-            'expenditure_percentile': self.expenditure_percentile(),
+            'salary_percentile': self.salary_percentile,
+            'expenditure_percentile': self.expenditure_percentile,
             'employee_salary_json': json.dumps(binned_employee_salaries),
             'data_year': 2017,
             'source_link': source_link,
         })
 
         return context
 
-    def median_entity_salary(self):
+    def get_median_entity_salary(self):
         q = Salary.objects.filter(Q(job__position__employer__parent=self.object) | Q(job__position__employer=self.object))
 
         results = q.all().aggregate(median=Percentile('amount', 0.5, output_field=FloatField()))
 
         return results['median']
 
+    # CacheMixin attribute
+    @property
+    def cache_prefix(self):
+        return str(self.object.id)
```

Comment:
@hancush Is this sufficiently unique?, i.e. is this unique to all objects or just to objects of the same model?